### DISCUSSION TITLE
[Feature Request] Classroom CLI: Cloning students' assignment repos should not fail if some already exist in the target directory.

### DISCUSSION THREAD

**Discussion by enricoschroeder** [community/community#55285] [GitHub Education]
### Hi everyone,

Hi Github Classroom team! 

I have a minor feature request that I think would make cloning students' assignment repos more streamlined: Do not abort when individual students' repos already exist in the target directory. 

The use-case is as follows: I clone my students' assignment repos with `gh classroom clone student-repos` once. If there are new student repos afterwards (e.g. because a student did not accept the assignment before), I would expect to be able to run the same command again to also clone the remaining repos.

However, the current implementation aborts cloning altogether since some of the repos already exist in the target directory:
```
Cloning into: <repo path>
2023/05/12 11:32:39 failed to run gh: fatal: destination path 'repo path' already exists and is not an empty directory.
failed to run git: exit status 128
. error: exit status 1
```
Therefore, I need to delete all of the local repos before being able to clone the new repos. So please at least include an optional flag that does not abort in this case. Thanks!

**Comment by rressler** [comment]
Agree with the request with a twist. 

Create a `gh classroom pull student-repos` command. This does what it says, it executes a pull request on repos which have already been cloned.

Instead of failing, provide the option to do a Git Pull using the new command.  LIkewise, is someone tries to pull but has not cloned, add the option to do a clone.

When students have a question on their assignments I clone the assignments so I can review their code. As they update their code, I just want to do a pull request instead of deleting the directory and recloning everyone.

In the past, I wrote my own scripts to do this but it would be much more useful if the CLI classroom extension added the new command.
